# HG changeset patch
# User aph
# Date 1584629637 0
#      Thu Mar 19 14:53:57 2020 +0000
# Node ID e96597ea4d672793ab205fd67009efed809fae11
# Parent  0daa6b52ba568246e8f930138fb36f8f04d9fadd
8241296: Segfault in JNIHandleBlock::oops_do()
Reviewed-by: duke

diff --git a/src/hotspot/share/runtime/thread.cpp b/src/hotspot/share/runtime/thread.cpp
--- a/src/hotspot/share/runtime/thread.cpp
+++ b/src/hotspot/share/runtime/thread.cpp
@@ -867,7 +867,9 @@
 }
 
 void Thread::oops_do(OopClosure* f, CodeBlobClosure* cf) {
-  active_handles()->oops_do(f);
+  if (active_handles() != NULL) {
+    active_handles()->oops_do(f);
+  }
   // Do oop for ThreadShadow
   f->do_oop((oop*)&_pending_exception);
   handle_area()->oops_do(f);
